index.js

```diff
@@ -25,56 +28,87 @@ function SonyBraviaTVAccessory(log, config) {
         .on('get', this.getOn.bind(this))
         .on('set', this.setOn.bind(this));
 
+    this.updateTimer();
+}
+
+SonyBraviaTVAccessory.prototype.runTimer = function() {
+    this.getState(function(err, isOn) {
+        if (err == null && isOn != this.isOn) {
+            this.log("State changed: %s", isOn ? "on" : "off");
+            this.service.getCharacteristic(Characteristic.On).updateValue(isOn);
+            this.isOn = isOn;
+        }
+    }.bind(this));
+}
+
+SonyBraviaTVAccessory.prototype.updateTimer = function() {
+    if (this.polling) {
+        clearTimeout(this.timer);
+        setTimeout(function() {
+            this.runTimer();
+            this.updateTimer();
+        }.bind(this), this.interval * 1000);
+    }
+}
+
+SonyBraviaTVAccessory.prototype.getState = function(callback) {
+  var postData = JSON.stringify({
+      method: 'getPowerStatus',
+      params: [],
+      id: 1,
+      version: '1.0'
+  });
+
+  request.post({
+      url: "http://" + this.ipaddress + "/sony/system",
+      headers: {
+          'X-Auth-PSK': this.psk
+      },
+      form: postData
+  }, function(err, response, body) {
+
+      if (!err && response.statusCode == 200) {
+          var json = JSON.parse(body);
+          var status = json.result[0].status;
+          var isOn = status == "active";
+          callback(null, isOn); // success
+      } else {
+          if (response != null) {
+              this.log("Error getting TV status (status code %s): %s", response.statusCode, err);
+              callback(err);
```

Comment:
Basicly this code was moved straight out of the getOn function into a new function, so that getOn and the timer could wrap it - I didn't put much effort in understanding its internal logic. If we get a 200 status from the TV, this code is not reached. I think, if we get a EHOSTUNREACH, (http) reponse is null and it should take the else branch below, which returns a null as error and a false as actual state. So this branch indicates some worse error? Though I wouldn't see anything wrong about considering the device off in that case.